### PR TITLE
Version of the DP checkout thank you page for order timeout scenarios

### PR DESCRIPTION
**Pull request by walaura**
## Why are you doing this?
AS A customer
I WANT you to tell me that my payment is still pending of it's not instantly processed
SO THAT I know my order may not have been processed yet

[**Trello Card**](https://trello.com/c/k1YYFvXn/2124-version-of-the-dp-checkout-thank-you-page-for-order-timeout-scenarios)

## Changes

* new `{| paymentStatus: 'success', isPending: true |}` payment result (i dont like it help)
* new `thankyou-pending` stage

## Screenshots

screenshot 2019-02-07 at 1 48 36 pm

### CODE REVIEW

File: assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ export type PaymentAuthorisation = StripeAuthorisation | PayPalAuthorisation | D
 // because the end of that checkout happens on the backend after the user is redirected to our site.
 export type PaymentResult
   = {| paymentStatus: 'success' |}
+  | {| paymentStatus: 'success', isPending: true |}
```
Comment (walaura) [inline assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js]: perhaps we should so something like `error: ErrorReason` below?

Comment (jacobwinch) [inline assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js]: Any reason you prefer this to: `{| paymentStatus: 'pending' |}`

I find it slightly misleading to mark the paymentStatus as success unless we know for sure that it has succeeded!

Comment (walaura) [reply assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js]: not really, shall we go with : pending?

Comment (jacobwinch) [reply assets/helpers/paymentIntegrations/newPaymentFlow/readerRevenueApis.js]: Now that we are using this to handle subscriptions as well as contributions, the following might be more appropriate:

```
export type CheckoutResult
= {| checkoutStatus: 'success' |}	
| {| checkoutStatus: 'success', subscriptionCreationPending: true |}
| {| checkoutStatus: 'failure', error: ErrorReason |};
```

Unlike contributions, the first payment for subscriptions products is not always taken immediately anyway (e.g. Digital Pack free trial).

**Pull request merged**